apps/src/music/blockly/blocks/simple2.js

```diff
@@ -397,19 +398,12 @@ export const repeatSimple2 = {
       'repeat_end',
       Blockly.Names.NameType.VARIABLE
     );
-    code += 'var ' + endVar + ' = ' + repeats + ';\n';
-    code +=
-      'for (var ' +
-      loopVar +
-      ' = 0; ' +
-      loopVar +
-      ' < ' +
-      endVar +
-      '; ' +
-      loopVar +
-      '++) {\n' +
-      branch +
-      '}\n';
+    code += `
+      var ${endVar} = ${repeats};
+      for (var ${loopVar} = 0; ${loopVar} < ${endVar} && __loopIterationsCount < ${MAX_LOOP_ITERATIONS_COUNT}; ${loopVar}++, __loopIterationsCount++) {
```

Comment:
Checking my understanding: seems like we could still end up with multiple nested loops with 999 iterations each, right? Could that still bring down our system, or do our other protections make that a non-issue?

Reply:
Given that `__loopIterationsCount` is global, the whole stack of nested loops should unwind once we hit the threshold.

Reply:
Ah right, missed that this was a global counter!